types/options.go

```diff
@@ -114,11 +120,48 @@ func loadDefaultStoreOptions() {
 
 // loadStoreOptions returns the default storage ops for containers
 func loadStoreOptions() (StoreOptions, error) {
-	storageConf, err := DefaultConfigFile()
+	baseConf, err := DefaultConfigFile()
 	if err != nil {
 		return defaultStoreOptions, err
 	}
-	return loadStoreOptionsFromConfFile(storageConf)
+
+	// Load the base config file
+	baseOptions, err := loadStoreOptionsFromConfFile(baseConf)
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	if _, err := os.Stat(defaultDropInConfigDir); err == nil {
+		// The directory exists, so merge the configuration from this directory
+		err = mergeConfigFromDirectory(&baseOptions, defaultDropInConfigDir)
+		if err != nil {
+			return defaultStoreOptions, err
+		}
+	} else if !os.IsNotExist(err) {
+		// There was an error other than the directory not existing
+		return defaultStoreOptions, err
+	}
+
+	return baseOptions, nil
+}
+
+func mergeConfigFromDirectory(baseOptions *StoreOptions, configDir string) error {
+	return filepath.Walk(configDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
```

Comment:
Nice update. Should check for file extension here.

Reply:
But a text file without a `.conf` or `.toml` can also contain a valid drop-in config. For any reason the content is not valid (irrespective of the extension) it will fail in `ReloadConfigurationFile` below while parsing the toml.

Reply:
Hmm. We support a file without an extension?

Reply:
as long as the content of the file is valid, it shouldn't matter, right? 

crio doesn't enforce the extension either - https://github.com/cri-o/cri-o/blob/256fda5ac98de1d67e26133d0b25df2a74e2ebd2/pkg/config/config.go#L776

Reply:
Usually there is a filter, but this is fine.

Reply:
one risk is temporary files written by editor and the like. It mostly only is an issue if we're reloading automatically. If it's manually triggered (like here) then the user usually has written and edited the editor before the reload happens...

Reply:
I think requiring a .conf or .toml extension could be a nice change both here and cri-o. @sohankunkerkar added a filter for the kubelet's drop-in work

Reply:
Thanks @rphillips @haircommander @rhatdan for this feedback. I have updated the code to select files with `.conf` extension only and also updated corresponding unit test to verify that.